### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by ryanwinchester**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+version: "3.7"
+services:
+
+  #PHP Service
+  app:
+    build:
+      args:
+        user: servidor
+        uid: 1000
```
Comment (ryanwinchester) [inline docker-compose.yml:9]:
```suggestion
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+RUN apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Install PHP extensions
+RUN docker-php-ext-install pdo_mysql mbstring exif pcntl bcmath gd
+
+# Get latest Composer
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+
+# Create system user to run Composer and Artisan Commands
+RUN useradd -G www-data,root -u $uid -d /home/$user $user
```
Comment (ryanwinchester) [inline Dockerfile:27]:
```suggestion
RUN useradd -b /var -UG www-data -s /usr/sbin/nologin --system $user
```

So you think just this is enough @dshoreman?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
+FROM php:7.4-fpm
+
+# Arguments defined in docker-compose.yml
+ARG user
+ARG uid
```
Comment (ryanwinchester) [inline Dockerfile:5]:
```suggestion
```

File: Dockerfile (hunk 2 above)
Comment (dshoreman) [reply Dockerfile]: Yep, should do it. Application code gets mapped to `/var/servidor` in the guest (or real server), so we set that as the home directory. That way, Composer dumps caches etc to `/var/servidor/.composer` to solve permission issues

**Comment by dshoreman** [comment]
Now that the installer is working properly with v0.10.0, I'm kind of curious how far it would get if we had a Dockerfile based on the official ubuntu 18.04 image since that's gonna be the easiest way to "mimic" a real server when using docker in development.

Volumes etc aside, theoretically it might *just* work (to some extent) simply by doing a `RUN ./setup.sh`... I may have to play about some after shopping and see what it does :thinking:

**Comment by ryanwinchester** [comment]
> Volumes etc aside, theoretically it might _just_ work (to some extent) simply by doing a `RUN ./setup.sh`... I may have to play about some after shopping and see what it does 🤔

You might have to add

```sh
COPY ./setup.sh ./setup.sh
COPY ./build ./build
```

To the dockerfile first unless those are already available because of the volume mounts. I'm on my phone and don't feel like looking at code on my phone, so dunno 😅😅😅

**Comment by ryanwinchester** [comment]
@dshoreman Actually, since the volume mounted is `- ./:/var/www`  you could probably do `RUN /var/www/setup.sh`?

However, should we switch that mount to ``- ./:/var/servidor`?

File: Dockerfile (hunk 2 above)
Comment (ryanwinchester) [reply Dockerfile]: @dshoreman should we switch volume mount from `./:/var/www` to `./:/var/servidor`?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# Get latest Composer
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+
+# Create system user to run Composer and Artisan Commands
+RUN useradd -b /var -UG www-data -s /usr/sbin/nologin --system $user
+RUN mkdir -p /home/$user/.composer && \
+    chown -R $user:$user /home/$user
+
+# Set working directory
+WORKDIR /var/www
```
Comment (ryanwinchester) [inline Dockerfile:31]: switch to `/var/servidor`?

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    container_name: servidor-app
+    restart: unless-stopped
+    tty: true
+    environment:
+      SERVICE_NAME: app
+      SERVICE_TAGS: dev
+    working_dir: /var/www
+    # Bind mount the laravel application data to /var/www
+    volumes:
+       - ./:/var/www
```
Comment (ryanwinchester) [inline docker-compose.yml:21]: switch also?

```suggestion
       - ./:/var/servidor
```

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      context: ./
+      dockerfile: Dockerfile
+    image: servidor
+    container_name: servidor-app
+    restart: unless-stopped
+    tty: true
+    environment:
+      SERVICE_NAME: app
+      SERVICE_TAGS: dev
+    working_dir: /var/www
```
Comment (ryanwinchester) [inline docker-compose.yml:18]: ?

```suggestion
    working_dir: /var/servidor
```

**Comment by ryanwinchester** [comment]
Ain't no way I'll have time for tinkering with this :(

**Pull request closed without merging**